src/metabase/app_db/cluster_lock.clj

```diff
@@ -22,6 +23,8 @@
   ;; was cancelled via timeout waiting to get the SELECT FOR UPDATE lock
   (or (instance? SQLIntegrityConstraintViolationException e)
       (instance? SQLIntegrityConstraintViolationException (ex-cause e))
+      ;; Postgres does just uses PSQLException, so we need to fall back to checking the message.
+      (str/includes? (ex-message e) "duplicate key value violates unique constraint \"metabase_cluster_lock_pkey\"")
```

Comment:
Since nested locks should catch their own collisions, we can get away without checking which lock.